pkg/cmd/api/api.go

```diff
@@ -265,10 +292,36 @@ func apiRun(opts *ApiOptions) error {
 		method = "POST"
 	}
 
+	var bodyWriter io.Writer = opts.IO.Out
+	var headersWriter io.Writer = opts.IO.Out
+	if opts.Silent {
+		bodyWriter = io.Discard
+	}
+	if opts.Verbose {
+		// httpClient handles output when verbose flag is specified.
+		bodyWriter = io.Discard
+		headersWriter = io.Discard
+	}
+
 	if opts.Paginate && !isGraphQL {
 		requestPath = addPerPage(requestPath, 100, params)
 	}
 
+	// Merge JSON arrays and objects if paginating without filtering or templating.
+	// MergePages retains compatibility with older versions but may be the default behavior with future major releases.
+	if opts.Paginate && opts.FilterOutput == "" && opts.Template == "" {
+		if !isGraphQL {
+			opts.merger = jsonmerge.NewArrayMerger()
+		} else if opts.MergePages {
+			opts.merger = jsonmerge.NewObjectMerger(bodyWriter)
+		}
+	}
+
+	if opts.merger == nil {
+		opts.merger = jsonmerge.NewNopMerger()
```

Comment:
I think we could do this unconditionally on line `311` so that it's always the default and then modified within the conditionals. Removes another `nil` check conditional.

Reply:
I considered doing that, but feels kinda dirty to alloc an object on the heap just to potentially throw it away. That said, it would be used in most cases, but this solves that as well. It is fragile, though. Which do you feel more strongly about preserving: memory (though, I suppose it is only a short-lived terminal process) or code maintainability?

Reply:
I hadn't considered that and I must admit I don't often give that kind of thing much thought. You nerd sniped me a bit here as I went off play with the compiler escape analysis 😅 . I generally favour maintainability, particularly that the heap allocation is a blip compared to the API request.

I suppose we could avoid this by returning the private type from the `New*Merger` functions rather than the `Merger` interface but really, I don't care much either way.

I feel 2/10 strongly about this, if we proceed with this PR so do whatever you want.